action/main.py

```diff
@@ -0,0 +1,52 @@
+import os
+import shlex
+import sys
+from pathlib import Path
+from subprocess import run, PIPE, STDOUT
+
+ACTION_PATH = Path(os.environ["GITHUB_ACTION_PATH"])
+ENV_PATH = ACTION_PATH / ".black-env"
+ENV_BIN = ENV_PATH / ("Scripts" if sys.platform == "win32" else "bin")
+OPTIONS = os.getenv("INPUT_OPTIONS", default="")
+SRC = os.getenv("INPUT_SRC", default="")
+BLACK_ARGS = os.getenv("INPUT_BLACK_ARGS", default="")
+VERSION = os.getenv("INPUT_VERSION", default="")
+
+# TODO: Uncomment these once https://github.com/actions/runner/issues/664 is resolved,
+# right now these cause more confusion than clarity :(
+# print("::group:: Setup & Install Black")
+
+run([sys.executable, "-m", "venv", str(ENV_PATH)], check=True)
+# print(f"Created virtual environment at `{ENV_PATH!s}`.")
+
+req = "black[colorama,python2]"
+if VERSION:
+    req += f"=={VERSION}"
+# TODO: remove output capturing / hiding once the log grouping works (because the logs
+# aren't out of order)
+pip_proc = run(
+    [str(ENV_BIN / "python"), "-m", "pip", "install", req],
+    stdout=PIPE,
+    stderr=STDOUT,
+    encoding="utf-8",
+)
+if pip_proc.returncode:
+    print(pip_proc.stdout)
+    print("::error::Failed to install Black.")
+    sys.exit(pip_proc.returncode)
+
+# print("::endgroup::")
+# print("::group:: Run Black")
```

Comment:
Want to cleanup or maybe comment that this is good for debugging or future use?

Reply:
They are pretty nice when the output isn't being mangled by a runner bug :( [example here](https://github.com/ichard26/testblackaction/runs/2606125162?check_suite_focus=true). I'm a fan of adding debug output when it's easy to hide them by default. I'll get rid of them since I honestly doubt the bug will be fixed anytime soon.